src/sentry/sentry_metrics/multiprocess.py

```diff
@@ -550,8 +570,13 @@ def submit(self, message: Message[KafkaPayload]) -> None:
             value=message.payload.value,
             on_delivery=self.callback,
         )
+        self.__producer.poll(0.0005)
```

Comment:
Also let's move this in the poll method, so that it will be called even for messages that do not trigger a batch to be processed. This would mean we would be dequeuing callbacks while we are building a batch as well and not waste time later.